### DISCUSSION TITLE
PCL Multi-Language Version - PCL 多语言版本

### DISCUSSION THREAD

**Discussion by tangge233** [Meloong-Git/PCL#4580] [讨论]
# This version of PCL has the ability of switching language!

## Basic Introduction

This version has no huge differences, compared with official release.We made some contributions to let it has the ability of switching language.

This is the official derivative version, which is subject to official oversight.

## Download

You can get the software updates by clicking "Check for update" button in the setting page.

[Go to release page](https://github.com/PCL-Community/PCL2-Language/releases/latest)

## Misc

Some data between the this version and the official version are not interoperable (Microsoft login data, some setting data)
Although the this version has more advanced functions, it may have some undiscovered problems.So please give us feedback by leaving a message here if you encounter any problems.

For more infomation about this version, please go to #4145 .

---

喂给搜索引擎的：`PCL2 怎么调语言` `PCL2 英文版` `PCL2 切换语言` `PCL2 自定义语言` `PCL2 语言文件`

**Comment by zkitefly** [comment]
如果要国外用户使用，我建议解决 PCL 只获取 BMCLAPI 的 Forge 列表

BMCLAPI 最近被打，把境外 ip 都禁了

**Reply by tangge233** [reply]
目前国外用户默认官方源优先

**Reply by zkitefly** [reply]
> 目前国外用户默认官方源优先

https://github.com/Hex-Dragon/PCL2/blob/f40a2990103ae85c34acb5d8d367ab1644aa7ca6/Plain%20Craft%20Launcher%202/Modules/Minecraft/ModDownload.vb#L1226-L1227

主要是看到了这个

**Comment by CreeperIsASpy** [comment]
离线登录的话我觉得没有问题，offline版本是mojang官方给的功能，只要你遵守eula，不商业化就没事的吧。

**Reply by 3gf8jv4dv** [reply]
The original intention of this design is to allow users who own the game to play without an Internet connection, rather than being able to play directly without owning the game.

Currently, many international launchers require users to log in to the Microsoft account that owns the game before they can launch the game in offline mode. Users can play the game directly without owning it, which conflicts with copyright laws in the United States and the European Union and is considered piracy.

**RELATED:**
- https://github.com/QuiltMC/quiltmc.org/issues/51
- https://github.com/PrismLauncher/PrismLauncher/pull/543

*Do not leave comments in the above issue / PR.*

Currently, it has been implemented in tangge233’s PR: in areas outside mainland China, you need to log in to the Microsoft account that owns the game before you can use offline login, otherwise the `--demo` parameter (Demo mode) will be added when launching the game.

---

这个设计（离线登录）的本意是让持有游戏的用户能够在没有网络的情况下游玩，而不是在未持有游戏的情况下可以直接游玩。

目前，许多国际化的启动器都要求用户登录持有游戏的正版账户，然后才能在离线模式下启动游戏。用户未持有游戏而可以直接游玩，这与美国和欧盟的版权法相冲突，被视为盗版行为。

**相关：**
- https://github.com/QuiltMC/quiltmc.org/issues/51
- https://github.com/PrismLauncher/PrismLauncher/pull/543

*不要在上述 issue / PR 中继续评论。*

目前 tangge233 的 PR 中已经实现：在中国大陆以外的区域需要登录正版账户后才可以使用离线登录，否则启动游戏时会添加 `--demo` 参数（演示模式）。

**Reply by CreeperIsASpy** [reply]
仔细想想，确实是这样的。
现在大多数国内玩家感觉买正版是做慈善一样。。
~~666我们大拆尼斯没有版权法是吧~~

**Reply by CreeperIsASpy** [reply]
~~你这个ai还能生成markdown？！给我介绍一下~~

**Reply by LinQingYuu** [reply]
>666我们大拆尼斯没有版权法是吧

挖个坟，还真没有《中华人民共和国版权法》这部法律，只有《中华人民共和国著作权法》、《中华人民共和国商标法》、《中华人民共和国专利法》

**Reply by HY916-cn** [reply]
> > 666我们大拆尼斯没有版权法是吧
> 
> 挖个坟，还真没有《中华人民共和国版权法》这条法律，只有《中华人民共和国著作权法》、《中华人民共和国商标法》、《中华人民共和国专利法》

欸！还真是欸

**Comment by zzr-shift** [comment]
版本设置-mod管理上面的显示有些过于冗杂了吧……
![4405aa59-0df5-476f-9335-f749012ba3da](https://github.com/user-attachments/assets/e959b1df-3563-47b7-96d8-8d6366fd7e9a)
我建议可以将这个MOD列表“-“后面的概述去掉，因为已经有按钮提示了，尤其是在MOD多一点的时候有点重合不好看

**Reply by tangge233** [reply]
那个是另一个分支的，而且已经合并了，暂时改不了了Orz

**Reply by 3gf8jv4dv** [reply]
> 那个是另一个分支的，而且已经合并了，暂时改不了了Orz

Open another one. :)

**Reply by tangge233** [reply]
#4636

**Reply by wyc-26** [reply]
> 版本设置-mod管理上面的显示有些过于冗杂了吧……
> 我建议可以将这个MOD列表“-“后面的概述去掉，因为已经有按钮提示了，尤其是在MOD多一点的时候有点重合不好看

此处讨论的是多语言开发版，请不要讨论其他内容，谢谢

**Comment by wyc-26** [comment]
@3gf8jv4dv Would you like to join our Weblate?
We've sent an invitation to your email.
<details><summary>Images</summary>

![image](https://github.com/user-attachments/assets/230f2194-2b09-480c-b034-b130034e38f9)

![image](https://github.com/user-attachments/assets/f70f4698-388e-4e85-b6e8-8d76e5225410)

</details>

**Reply by 3gf8jv4dv** [reply]
Sure, but I need some time to get used to Weblate. I will deal with this invitation later.

**Comment by HY916-cn** [comment]
如何参与翻译？我和我的朋友乐意效劳（当然也有国外的朋友

**Reply by 3gf8jv4dv** [reply]
Please visit https://weblate.tangge233.cn/

**Reply by HY916-cn** [reply]
> 请访问 https://weblate.tangge233.cn/

无权限

**Reply by 3gf8jv4dv** [reply]
> 无权限

Read the *Translation Guide* at the following link

https://weblate.tangge233.cn/projects/PCL/

**Reply by tangge233** [reply]
您好，很抱歉不能直接编辑给您带来了困扰。翻译站为了避免一些修改不达预期以及一些恶意的修改，目前我们只开放译文建议，而不能直接翻译。

**Reply by HY916-cn** [reply]
> 您好，很抱歉不能直接编辑给您带来了困扰。翻译站为了避免一些修改不达预期以及一些恶意的修改，目前我们只开放译文建议，而不能直接翻译。

是否有办法加入你们的团队或者获得翻译权限

**Reply by tangge233** [reply]
emmmm
之后社区会考虑下直接开放翻译权限

**Reply by tangge233** [reply]
已经开放直接翻译权限，请注意已更新的翻译指南

**Reply by HY916-cn** [reply]
> 已经开放直接翻译权限，请注意已更新的翻译指南

Thx!

**Reply by HY916-cn** [reply]
> emmmm 之后社区会考虑下直接开放翻译权限

我想到一个方法，我和我的朋友以库中的语言文件作参考，自行翻译成目标语言，后将语言xaml文件上交并审核后直接加入下一版本（？）若该方法可行，我们将开工！

**Reply by tangge233** [reply]
之前没用 weblate 时差不多就是这么弄的，直接编辑源文件进行翻译。如果你想添加新的语言的话，需要把文件提交到我仓库的 language 分支。目前 PCL 在未来一段时间内不会合并我的 PR，但是多语言版本会保持更新。

**Reply by tangge233** [reply]
> 所以翻译指南在哪里没找到awa

翻译站的 PCL 项目上面的公告就是，不过那是关于在翻译站进行翻译的简单指导

**Reply by HY916-cn** [reply]
> > 所以翻译指南在哪里没找到awa
> 
> 翻译站的 PCL 项目上面的公告就是，不过那是关于在翻译站进行翻译的简单指导

已经获得翻译权限了，感谢您！

**Reply by HY916-cn** [reply]
> > 所以翻译指南在哪里没找到awa
> 
> 翻译站的 PCL 项目上面的公告就是，不过那是关于在翻译站进行翻译的简单指导

请问是否可以增加斯洛伐克语？

**Reply by tangge233** [reply]
> 请问是否可以增加斯洛伐克语？

语言代码是 sk_SK，稍后我加入切换支持，你可以直接开翻

**Reply by HY916-cn** [reply]
> > 请问是否可以增加斯洛伐克语？
> 
> 语言代码是 sk_SK，稍后我加入切换支持，你可以直接开翻

⚡ 稍 后 ⚡

**Reply by tangge233** [reply]
> ⚡ 稍 后 ⚡

抱歉，今天比较忙。我已经上传了语言文件（英文为模板）。

**Reply by HY916-cn** [reply]
> > ⚡ 稍 后 ⚡
> 
> 抱歉，今天比较忙。我已经上传了语言文件（英文为模板）。

thx 正在翻译

**Comment by iPodtouch6557** [comment]
提供一个小思路（仅为参考之用！）
https://b23.tv/BojkTjb

**Reply by Chiloven945** [reply]
这个没有任何联系的……

**Reply by Deep-Dark-Forest** [reply]
> 这个没有任何联系的……

是的，PCL 本身又不是游戏

**Comment by kong-que-dnf123** [comment]
无法访问https://weblate.tangge233.cn/projects/PCL/ 因为Error1033……
![image](https://github.com/user-attachments/assets/0245d131-6b98-463e-8f57-543b7fc01852)

**Reply by Ad-closeNN** [reply]
@tangge233 在维护，等一会就好了
![image](https://github.com/user-attachments/assets/e6329f94-f9c3-4b5c-b800-2d5ef83c96c6)

**Comment by wtommy932** [comment]
目前weblate似乎无法注册...
<img width="2560" height="1528" alt="image" src="https://github.com/user-attachments/assets/1fb35e26-7a6e-4de6-9bfb-56e2db74dee7" />
希望早日修复

**Comment by yichenjj** [comment]
我想提供完整的更人性化西班牙语翻译，我该怎么做

**Reply by Chiloven945** [reply]
你可以在 [tangge233/PCL 的 Language 分支](https://github.com/tangge233/PCL2/tree/Language) 下找到文件 [/Plain Craft Launcher 2/Resources/Language/es-ES.xaml](https://github.com/tangge233/PCL2/blob/Language/Plain%20Craft%20Launcher%202/Resources/Language/es-ES.xaml)，这是存储西班牙语翻译的地方。

使用编辑器打开文件后编辑翻译文本即可，请务必参考 [PCL i18n & l10n 项目指南](https://github.com/tangge233/PCL2/blob/Language/CONTRIBUTING.md) 了解翻译流程与注意事项。

随后将仓库 fork 到本地后修改文件并向分支 Language 提交 Pull Request。如果你不会使用 Git 可以在这发送文件的链接，我们可以帮你（

**Reply by yichenjj** [reply]
好的谢谢

**Reply by yichenjj** [reply]
> 你可以在 [tangge233/PCL 的 Language 分支](https://github.com/tangge233/PCL2/tree/Language) 下找到文件 [/Plain Craft Launcher 2/Resources/Language/es-ES.xaml](https://github.com/tangge233/PCL2/blob/Language/Plain%20Craft%20Launcher%202/Resources/Language/es-ES.xaml)，这是存储西班牙语翻译的地方。
> 
> 使用编辑器打开文件后编辑翻译文本即可，请务必参考 [PCL i18n & l10n 项目指南](https://github.com/tangge233/PCL2/blob/Language/CONTRIBUTING.md) 了解翻译流程与注意事项。
> 
> 随后将仓库 fork 到本地后修改文件并向分支 Language 提交 Pull Request。如果你不会使用 Git 可以在这发送文件的链接，我们可以帮你（

不让我把文件发给你好像是文件格式不对
<img width="874" height="118" alt="image" src="https://github.com/user-attachments/assets/7389bb88-367e-4246-b0fb-0a907d50588f" />

**Reply by LinQingYuu** [reply]
你可以用压缩软件压成 zip 然后发上来

**Reply by yichenjj** [reply]
> 你可以用压缩软件压成 zip 然后发上来

[es-ES.zip](https://github.com/user-attachments/files/26143756/es-ES.zip)
帮我发下呗我不会

**Comment by JellyLight** [comment]
？
